Source/AwsCommonRuntimeKit/crt/ByteBuffer.swift

```diff
@@ -2,285 +2,68 @@
 //  SPDX-License-Identifier: Apache-2.0.
 
 import struct Foundation.Data
-import class Foundation.InputStream
-import class Foundation.FileManager
-import class Foundation.FileHandle
-import class Foundation.OutputStream
-import struct Foundation.URL
 import AwsCIo
-import AwsCCommon
 import AwsCCal
-#if os(Linux)
-     import Glibc
- #else
-     import Darwin
- #endif
 
-// swiftlint:disable identifier_name superfluous_disable_command
-// Todo: this is being used. Refactor
-public class ByteBuffer: Codable {
+public class ByteBuffer {
+
+    private var data: Data
+    private var currentIndex: Data.Index
```

Comment:
Why not a simple Int? https://forums.swift.org/t/data-popfirst-removefirst-adjust-indices/35965/5